### DISCUSSION TITLE
Add rate limiting options for `@wagmi/cli` block explorer plugins

### DISCUSSION THREAD

**Discussion by tyko0x** [wevm/wagmi#4122] [Ideas]
Block explorers like etherscan often have strict rate limiting: https://etherscan.io/apis. For example, under the free etherscan plan, if you are fetching more than 5 ABIs, then the plugins errors out since the rate limit is 5 calls per second.

I suggest adding some rate limiting options - such as `limit` and `interval` options.

**Comment by kasparkallas** [comment]
Yep, Etherscan is very script about the calls. Even paid version is just 10 requests per second.

Workaround is to put this somewhere in the `etherscan.js` in `node_modules` for `@wagmi/cli` & `../dist/esm/plugins`:
```
const json = await globalThis
    .fetch(buildUrl({ ...options, action: 'getsourcecode' }))
    .then((res) => res.json())
    .then((json) => new Promise((resolve) => setTimeout(() => resolve(json), 1000))); // This...
```
